torch/nn/modules/activation.py

```diff
@@ -777,8 +777,9 @@ def forward(self, query, key, value, key_padding_mask=None,
             be ignored by the attention. This is an binary mask. When the value is True,
             the corresponding value on the attention layer will be filled with -inf.
         need_weights: output attn_output_weights.
-        attn_mask: mask that prevents attention to certain positions. This is an additive mask
-            (i.e. the values will be added to the attention layer).
+        attn_mask: 2D or 3D mask that prevents attention to certain positions. This is an additive mask
+            (i.e. the values will be added to the attention layer). A 2D mask will be unsqueezed
```

Comment:
There should be a cleaner way of writing this instead of "A 2d mask will be unsqueezed" which exactly describes the code. 

Really what we're supporting here is broadcasting the mask over the batches if it is of lower dimension than the input?